money-core/src/main/scala/com/comcast/money/core/internal/SpanLocal.scala

```diff
@@ -46,36 +46,33 @@ object SpanLocal extends SpanContext {
   override def current: Option[Span] = Option(threadLocalCtx.get).flatMap(_.headOption)
 
   override def push(span: Span): Unit =
-    if (span != null)
+    if (span != null) {
       Option(threadLocalCtx.get) match {
-        case Some(stack) =>
-          stack.push(span)
-          setSpanMDC(Some(span.info.id))
-          setSpanNameMDC(Some(span.info.name))
+        case Some(current) =>
+          threadLocalCtx.set(span :: current)
         case None =>
-          threadLocalCtx.set(new Stack[Span]())
-          push(span)
+          threadLocalCtx.set(List(span))
       }
+      setSpanMDC(Some(span.info.id))
+      setSpanNameMDC(Some(span.info.name))
+    }
 
   override def pop(): Option[Span] =
-    Option(threadLocalCtx.get).map {
-      stack =>
-        // remove the current span in scope for this thread
-        val spanId = stack.pop()
-
-        // rolls back the mdc span to the parent if present, if none then it will be cleared
-        val head = stack.headOption
+    threadLocalCtx.get match {
+      case span :: rest =>
+        threadLocalCtx.set(rest)
+        val head = rest.headOption
         setSpanMDC(head.map(_.info.id))
         setSpanNameMDC(head.map(_.info.name))
-
-        spanId
+        Some(span)
```

Comment:
probably could improve this slightly...

```scala
rest.headOption.foreach { head => 
  setSpanMDC(head.info.id)
  setSpanNameMDC(head.info.name)
}
Some(span)
```

Reply:
The only issue with this is that I'd still want to call those `setSpan*` methods with `None` if there is no `Span` so that the MDC keys are cleared.